roles/kubernetes/control-plane/tasks/kubeadm-setup.yml

```diff
@@ -166,22 +166,37 @@
     - not kube_external_ca_mode
 
 - name: Kubeadm | Initialize first control plane node
-  command: >-
-    timeout -k {{ kubeadm_init_timeout }} {{ kubeadm_init_timeout }}
-    {{ bin_dir }}/kubeadm init
-    --config={{ kube_config_dir }}/kubeadm-config.yaml
-    --ignore-preflight-errors={{ kubeadm_ignore_preflight_errors | join(',') }}
-    --skip-phases={{ kubeadm_init_phases_skip | join(',') }}
-    {{ kube_external_ca_mode | ternary('', '--upload-certs') }}
-  register: kubeadm_init
-  # Retry is because upload config sometimes fails
-  retries: 3
-  until: kubeadm_init is succeeded or "field is immutable" in kubeadm_init.stderr
   when: inventory_hostname == first_kube_control_plane and not kubeadm_already_run.stat.exists
-  failed_when: kubeadm_init.rc != 0 and "field is immutable" not in kubeadm_init.stderr
   environment:
     PATH: "{{ bin_dir }}:{{ ansible_env.PATH }}"
-  notify: Control plane | restart kubelet
+  block:
+    - name: Kubeadm | Set kubeadm init command for first control plane node
+      set_fact:
+        kubeadm_init_first_cp_cmd: >-
+          timeout -k {{ kubeadm_init_timeout }} {{ kubeadm_init_timeout }}
+          {{ bin_dir }}/kubeadm init
+          --config={{ kube_config_dir }}/kubeadm-config.yaml
+          --ignore-preflight-errors={{ kubeadm_ignore_preflight_errors | join(',') }}
+          --skip-phases={{ kubeadm_init_phases_skip | join(',') }}
+          {{ kube_external_ca_mode | ternary('', '--upload-certs') }}
```

Comment:
If we do that, this should be  a var and not a set_fact

Reply:
I will fix this.

Reply:
Fixed.